querydsl-codegen/src/main/java/com/querydsl/codegen/EntityType.java

```diff
@@ -66,14 +75,13 @@ public EntityType(Type type) {
      *
      * @param type
      * @param superTypes
+     * @param variableNameFunction
      */
-    public EntityType(Type type, Set<Supertype> superTypes) {
+    public EntityType(Type type, Set<Supertype> superTypes, Function<EntityType, String> variableNameFunction) {
```

Comment:
Maybe getUncapSimpleName could be deprecated and the transformation done outside of EntityType?

```java
String variableName = variableNameFunction(type.getSimpleName());
if (JavaSyntaxUtils.isReserved(name)) {
    variableName = variableName + "$";
}
```